hrqb/utils/__init__.py

```diff
@@ -3,15 +3,52 @@
 import datetime
 
 import click
+import pandas as pd
+import us  # type: ignore[import-untyped]
+from dateutil.parser import parse as date_parser  # type: ignore[import-untyped]
 
 
 def today_date() -> datetime.date:
     return datetime.datetime.now(tz=datetime.UTC).date()
 
 
+def normalize_date(date: str | datetime.datetime) -> str | None:
+    if date is None or date == "" or pd.isna(date):
+        return None
+    if isinstance(date, str):
+        date = date_parser(date)
+    if isinstance(date, datetime.datetime):
+        return date.strftime("%Y-%m-%d")
+    return None
```

Comment:
Hmm, when viewing this code locally, `Pylance` indicates "Code is unreachable" for the final `return None`. Should there be some error handling for dates that cannot be parsed?

Perhaps rewrite to:

```
import logging

def normalize_date(date: str | datetime.datetime) -> str | None:
    if date is None or date == "" or pd.isna(date):
        return None
    if isinstance(date, str):
        try:
            date = date_parser(date)
        except ParserError:
            logging.warning("Unable to parse date from '{date}'")
    if isinstance(date, datetime.datetime):
        return date.strftime("%Y-%m-%d")
    return None
```

Reply:
This utility function has been a headache, and @ehanson8 commented on it too.  I think this is a good change.  You're (and pylance) are right, that if the string is not parsable, it would raise an exception so we'd never hit that `return None` (because if it _did_ parse it, then we'd return on the next block).  Updating!